imports/plugins/core/checkout/client/components/completedOrder.js

```diff
@@ -18,95 +20,219 @@ import AddEmail from "./addEmail";
  * @property {Booleam} isProfilePage - A boolean value that checks if current page is user profile page
  * @return {Node} React node containing the top-level component for displaying the completed order/receipt page
  */
-const CompletedOrder = ({ order, orderId, shops, orderSummary, paymentMethods, handleDisplayMedia, isProfilePage }) => {
-  if (!order) {
-    return (
-      <Components.NotFound
-        i18nKeyTitle="order.notFound"
-        icon="fa fa-barcode"
-        title="Order Not Found"
-      />
-    );
+class CompletedOrder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      order: props.order,
+      orderId: props.orderId,
+      shops: props.shops,
+      orderSummary: props.orderSummary,
+      paymentMethods: props.paymentMethods,
+      handleDisplayMedia: props.handleDisplayMedia,
+      isProfilePage: props.isProfilePage,
+      isCancelled: false
+    };
   }
+   cancelOrder = event => {
+     event.preventDefault();
+     Alerts.alert({
+       title: "Cancel Order",
+       type: "question",
+       text: "You're about to cancel this order",
+       showCancelButton: true
+     }, (isConfirmed) => {
+       if (isConfirmed) {
+         const { order } = this.props;
+         const { email } = order;
+         const amount = order.billing[0].invoice.total;
+         const transaction = {
+           amount: Number(amount),
+           to: email,
+           from: email,
+           transactionType: "credit"
+         };
+         Meteor.call("orders/cancelOrder", order, true);
+         Meteor.call("wallet/update-balance",
+           transaction,
+           (err, res) => {
+             if (res === 2) {
```

Comment:
What does 2 and 1 mean here and on line 61?

Reply:
That figure `2` is actually supposed to be `0`. The **wallet/update-balance** method returns 1 for success and 0 for error. I have applied the changes accordingly.
I have also corrected the typo and removed the commented codes.